diff --git a/node_modules/buefy/src/components/modal/Modal.vue b/node_modules/buefy/src/components/modal/Modal.vue
index 61ee674..9185170 100644
--- a/node_modules/buefy/src/components/modal/Modal.vue
+++ b/node_modules/buefy/src/components/modal/Modal.vue
@@ -73,6 +73,14 @@ export default {
             type: [String, Number],
             default: 960
         },
+        appendToBody: {
+            type: Boolean,
+            default: true,
+        },
+        appendTo: {
+            type: HTMLElement,
+            default: null
+        },
         hasModalCard: Boolean,
         animation: {
             type: String,
@@ -232,10 +240,10 @@ export default {
         close() {
             this.$emit('close')
             this.$emit('update:active', false)
+            this.isActive = false
 
             // Timeout for the animation complete before destroying
-            if (this.programmatic) {
-                this.isActive = false
+            if (this.programmatic && this.destroyOnHide) {
                 setTimeout(() => {
                     this.$destroy()
                     removeElement(this.$el)
@@ -243,6 +251,14 @@ export default {
             }
         },
 
+        open() {
+            if (!this.destroyOnHide) {
+                this.$emit('open')
+                this.$emit('update:active', true)    
+                this.isActive = true
+            }
+        },
+
         /**
         * Keypress event that is bound to the document.
         */
@@ -283,7 +299,16 @@ export default {
     beforeMount() {
         // Insert the Modal component in body tag
         // only if it's programmatic
-        this.programmatic && document.body.appendChild(this.$el)
+        if (this.programmatic) {
+            if (this.appendTo && this.appendTo instanceof HTMLElement) {
+                this.appendTo.appendChild(this.$el)
+            } else if (this.appendToBody) {
+                document.body.appendChild(this.$el)
+            } else {
+                this.$parent.$el.appendChild(this.$el)
+            }
+        }
+        // this.programmatic && document.body.appendChild(this.$el)
     },
     mounted() {
         if (this.programmatic) this.isActive = true
